src/bin/cql-stress-cassandra-stress/settings/mod.rs

```diff
@@ -0,0 +1,143 @@
+    pub command: Command,
+    pub params: CommandParams,
+}
+
+impl CassandraStressSettings {
+    fn new(command: Command, params: CommandParams) -> Self {
+        Self { command, params }
+    }
+
+    pub fn print_settings(&self) {
+        println!("******************** Stress Settings ********************");
+        self.params.print_settings(&self.command);
+        println!();
+    }
+}
+
+pub enum CassandraStressParsingResult {
+    Special,
+    Workload(Box<CassandraStressSettings>),
+}
+
+type ParsePayload<'a> = HashMap<&'a str, Vec<&'a str>>;
+
+/// Groups the commands/options and their corresponding parametes.
+///
+/// cassandra-stress accepts CLI args of the following pattern:
+/// ./cassandra-stress COMMAND [command_param...] [OPTION [option_param...]...]
+fn prepare_parse_payload(args: &[String]) -> Result<(&str, ParsePayload)> {
+    let mut cl_args: ParsePayload = HashMap::new();
+    let mut current: &str = "";
+    let mut cmd: &str = "";
+    for (i, arg) in args.iter().enumerate() {
+        let arg = arg.as_ref();
+        if i == 0 {
+            cmd = arg;
+        }
+        if i == 0 || arg.starts_with('-') {
+            anyhow::ensure!(
+                !cl_args.contains_key(arg),
+                "{} is defined multiple times. Each option/command can be specidied at most once.",
+                arg
+            );
+            current = arg;
+            cl_args.insert(arg, vec![]);
+            continue;
+        }
+
+        let params = cl_args.get_mut(current).unwrap();
+        params.push(arg);
+    }
+
+    Ok((cmd, cl_args))
+}
+
+lazy_static! {
+    // Removes whitespaces before characters: ,=()
+    static ref REPLACE_1: Regex = Regex::new(r"\s+([,=()])").unwrap();
+    // Removes whitespaces after characters: ,=(
+    static ref REPLACE_2: Regex = Regex::new(r"([,=(])\s+").unwrap();
+    // Splits the resulting arguments by whitespaces.
+    static ref SPLIT_REGEX: Regex = Regex::new(r" +").unwrap();
+}
+
+/// Removes the unnecessary whitespaces from the arguments,
+/// and then splits the arguments that contain whitespaces.
+/// For example when user passes following command (cassandra-stress accepts such command):
+/// ./cassandra-stress read -rate 'threads=80 throttle=8000/s'
+///
+/// Note that 'threads=80 throttle=8000/s' will be treated as a single string,
+/// so we need to split this into two separate parameters.
+fn repair_params<I, S>(args: I) -> Vec<String>
+where
+    I: Iterator<Item = S>,
+    S: AsRef<str>,
+{
+    // Concat to single string.
+    let s = args
+        .fold((true, String::from("")), |(is_first, mut acc), next| {
+            if !is_first {
+                acc += " ";
+            }
+            acc += next.as_ref();
+            (false, acc)
+        })
+        .1;
+    let replaced = REPLACE_1.replace_all(&s, "$1");
+    let replaced = REPLACE_2.replace_all(&replaced, "$1");
+    SPLIT_REGEX.split(&replaced).map(&str::to_owned).collect()
+}
```

Comment:
A small test of this function would be appreciated - this would also make it easier to understand what this function does (it took me a while to understand why `,` is in `REPLACE_1`, `REPLACE_2`).